lib/Reader/readerir.cpp

```diff
@@ -7666,97 +7706,186 @@ Type *GenIR::Vector4Ty;
+    return (IRNode *)LLVMBuilder->CreateMul(Vector1, Vector2);
+  } else {
+    return 0;
+  }
 }
 
-IRNode *GenIR::vectorDiv(IRNode *Vector1, IRNode *Vector2) {
-  assert(((Value *)Vector1)
-             ->getType()
-             ->getVectorElementType()
-             ->isFloatingPointTy());
-  assert(((Value *)Vector2)->getType() == ((Value *)Vector1)->getType());
-  return (IRNode *)LLVMBuilder->CreateFDiv(Vector1, Vector2);
+IRNode *GenIR::vectorDiv(IRNode *Vector1, IRNode *Vector2,
+                         CORINFO_CLASS_HANDLE Class) {
+  assert(Vector2->getType() == Vector1->getType());
+  if (Vector1->getType()->getVectorElementType()->isFloatingPointTy()) {
+    return (IRNode *)LLVMBuilder->CreateFDiv(Vector1, Vector2);
+  } else if (Vector1->getType()->getVectorElementType()->isIntegerTy()) {
+    int Length = 0;
+    bool IsGeneric = false;
+    bool IsSigned = false;
+    getBaseTypeAndSizeOfSIMDType(Class, Length, IsGeneric, IsSigned);
+    if (IsSigned) {
+      return (IRNode *)LLVMBuilder->CreateSDiv(Vector1, Vector2);
+    } else {
+      return (IRNode *)LLVMBuilder->CreateUDiv(Vector1, Vector2);
+    }
+  } else {
+    return 0;
+  }
 }
 
 IRNode *GenIR::vectorEqual(IRNode *Vector1, IRNode *Vector2) {
-  assert(((Value *)Vector1)
-             ->getType()
-             ->getVectorElementType()
-             ->isFloatingPointTy());
-  assert(((Value *)Vector2)->getType() == ((Value *)Vector1)->getType());
-  return (IRNode *)LLVMBuilder->CreateFCmpOEQ(Vector1, Vector2);
+  assert(Vector2->getType() == Vector1->getType());
+  if (Vector1->getType()->getVectorElementType()->isFloatingPointTy()) {
+    return (IRNode *)LLVMBuilder->CreateFCmpOEQ(Vector1, Vector2);
+  }
+  if (Vector1->getType()->getVectorElementType()->isIntegerTy()) {
+    return (IRNode *)LLVMBuilder->CreateICmpEQ(Vector1, Vector2);
+  }
+  return 0;
 }
 
 IRNode *GenIR::vectorNotEqual(IRNode *Vector1, IRNode *Vector2) {
-  assert(((Value *)Vector1)
-             ->getType()
-             ->getVectorElementType()
-             ->isFloatingPointTy());
-  assert(((Value *)Vector2)->getType() == ((Value *)Vector1)->getType());
-  return (IRNode *)LLVMBuilder->CreateFCmpONE(Vector1, Vector2);
+  assert(Vector2->getType() == Vector1->getType());
+  if (Vector1->getType()->getVectorElementType()->isFloatingPointTy()) {
+    return (IRNode *)LLVMBuilder->CreateFCmpONE(Vector1, Vector2);
+  }
+  if (Vector1->getType()->getVectorElementType()->isIntegerTy()) {
+    return (IRNode *)LLVMBuilder->CreateICmpNE(Vector1, Vector2);
+  }
+  return 0;
+}
+
+IRNode *GenIR::vectorMax(IRNode *Vector1, IRNode *Vector2) {
+  assert(Vector2->getType() == Vector1->getType());
+  if (Vector1->getType()->getVectorElementType()->isFloatingPointTy()) {
+    std::vector<Type *> Types;
+    Types.push_back(Vector1->getType());
+    llvm::Function *Func = 0;
+    Func = Intrinsic::getDeclaration(JitContext->CurrentModule,
```

Comment:
Hm, I see the bit in the docs that mentions vectors now. I was confused by the fact that vector types are missing from the list of signatures. Given #733, it might be interesting to try generating the suggested IR above in any case.